torch/csrc/jit/script/compiler.cpp

```diff
@@ -474,6 +489,87 @@ struct to_ir {
     }
   }
 
+  void emitFor(const For& stmt) {
+    // For now, we only support range loops. e.g. for i in range(3): ...
+
+    auto targets = stmt.targets();
+    auto itrs = stmt.itrs();
+    auto body = stmt.body();
+
+    // itrs must consist of a single Apply node
+    if (stmt.itrs().size() != 1) {
+      throw ErrorReport(stmt)
+          << "List of iterables is not supported currently.";
+    }
+    if (itrs[0].kind() != TK_APPLY) {
+      throw ErrorReport(stmt)
+          << "Non-range for loops are currently not supported.";
+    }
+
+    Apply range_iterator = Apply(itrs[0]);
+    if (range_iterator.callee().kind() != TK_VAR) {
+      throw ErrorReport(stmt)
+          << "Non-range for loops are currently not supported.";
+    }
+
+    {
+      Var var = Var(range_iterator.callee());
+      if (var.name().name() != "range") {
+        throw ErrorReport(stmt)
+            << "Non-range for loops are currently not supported.";
+      }
+    }
+
+    int max_trip_count_val;
+    {
+      List<Expr> args = range_iterator.inputs();
+      // TODO: start, stop, step loop
+      if (args.size() != 1) {
+        throw ErrorReport(stmt)
+            << "range() expects one argument but got" << args.size();
+      }
+
+      if (args[0].kind() != TK_CONST) {
+        throw ErrorReport(stmt) << "Argument to range() must be a constant";
```

Comment:
We don't really need this right? You could just `emitExpr` and use the result as the iterator value